syntax = "proto3";

package ai.wombo.tao.miner;

import "tensor/protos/inputs.proto";
import "neuron/protos/neuron.proto";

message OutputScore {
  float score = 1;
}

message ValidatorUserRequest {
  ai.wombo.tao.inputs.GenerationRequestInputs inputs = 1;
  bool watermark = 2;
  optional uint32 miner_uid = 3;
}

message ValidatorGenerationResponse {
  bytes image = 1;
  uint32 miner_uid = 2;
  float generation_time = 3;
}

service ForwardingValidator {
  rpc ScoreOutput(ai.wombo.tao.neuron.MinerGenerationResponse) returns (OutputScore);
  rpc Generate(ValidatorUserRequest) returns (ValidatorGenerationResponse);
}
